baseband/vdif/tests/test_vdif.py

```diff
@@ -111,6 +112,78 @@ def test_header(self):
         assert abs(header5.time - header.time - 1.*u.s) < 1.*u.ns
         assert header5['frame_nr'] == header['frame_nr']
 
+    def test_custom_header(self):
+        # Custom header with an EDV that already exists
+        with pytest.raises(ValueError):
+            @six.add_metaclass(vdif.header._VDIFHeaderRegistry)
+            class VDIFHeaderY(vdif.header.VDIFBaseHeader):
+                _edv = 4
+        # Custom header with an invalid EDV
+        with pytest.raises(ValueError):
+            @six.add_metaclass(vdif.header._VDIFHeaderRegistry)
+            class VDIFHeaderZ(vdif.header.VDIFBaseHeader):
+                _edv = None
+        # Custom header that fails to subclass VDIFHeader
+        with pytest.raises(TypeError):
```

Comment:
I'd remove this test - this is not our worry (nobody should use the metaclass; it should not be exported).